### ISSUE TITLE
Add `mask_token_ids` and `update_token_ids` API

### ISSUE THREAD

**Issue by Dan-wanna-M** [Dan-wanna-M/kbnf#7]
Some users might want to mask an array of token ids(e.g. from `top_p`, `top_k`) rather than the whole logits. We probably need the caller to provide an output buffer considering how the FFI works.

**Comment by Dan-wanna-M** [comment]
This will essentially stop cache from functioning. Probably should be implemented after eager regex cache.
